mempool/mempool.go

```diff
@@ -692,6 +694,19 @@ func (mp *TxPool) maybeAcceptTransaction(tx *btcutil.Tx, isNew, rateLimit, rejec
 		return missingParents, nil
 	}
 
+	// Don't allow the transaction into the mempool unless its sequence
+	// lock is active, meaning that it'll be allowed into the next block
+	// with respect to its defined relative lock times.
+	sequenceLock, err := mp.cfg.CalcSequenceLock(tx, utxoView)
+	if err != nil {
+		return nil, err
```

Comment:
Like other paths that involve return errors from `blockchain` this needs to be converted via `chainRuleError` if it's a `blockchain.RuleError`.  This is important so an appropriate reject message is generated and delivered back to the sending peer.

Here is the diff:

```diff
diff --git a/mempool/mempool.go b/mempool/mempool.go
index a9630a4..0a1ea9e 100644
--- a/mempool/mempool.go
+++ b/mempool/mempool.go
@@ -705,6 +705,9 @@ func (mp *TxPool) maybeAcceptTransaction(tx *btcutil.Tx, isNew, rateLimit, rejec
        // with respect to its defined relative lock times.
        sequenceLock, err := mp.cfg.CalcSequenceLock(tx, utxoView)
        if err != nil {
+               if cerr, ok := err.(blockchain.RuleError); ok {
+                       return nil, chainRuleError(cerr)
+               }
                return nil, err
        }
```